shader/svgf_atrous.comp

```diff
@@ -1,185 +1,140 @@
-                    p.z
-                );
-
-                if (xx == 00 && yy == 00)
-                    continue;
-
-                float w = float(all(greaterThanEqual(o.xy, ivec2(0)))
-                            && all(lessThan(o.xy, control.size)));
-
-                vec4 color_o = sample_color(o);
-                float lum_color_o = luminance(color_o.rgb);
-                float dist_l = abs(color_luminance_center - lum_color_o);
-
-                vec4 specular_o = sample_specular(o);
-
-                vec3 normal = unpack_gbuffer_normal(imageLoad(in_normal, o).xy);
-                vec3 position = imageLoad(in_pos, o).xyz;
-                vec3 diffp = position - position_center;
-                float distp = dot(diffp, diffp);
-
-                w *= pow(max(0.0, dot(normal_center, normal)), 128.0);  // normal based weight
-                w *= any(isnan(position)) ? 0.0 : exp(-distp);          // position based weight
-                /*
-                // Alternate, more scale-independent version:
-                vec3 tangent = normalize(position_center - position);
-                w *=
-                    clamp(1.0f-abs(dot(tangent, normal_center))*3.0f, 0.0, 1.0) *
-                    clamp(dot(normal, normal_center)*2.0f-1.0f, 0.0f, 1.0f);
-                */
-                w *= max(exp(-dist_l * dist_l * sigma_l), 0.0001f); // luminance based weight
-                // w *= kernel[kernel_index];
-                w *= wavelet_kernel[abs(xx)][abs(yy)];
-                ++kernel_index;
-
-                if (control.level < 2)
+                const vec4 color_q = (control.level & 1) == 0 ? imageLoad(diffuse_ping, q) : imageLoad(diffuse_pong, q);
+                const vec4 spec_q =  (control.level & 1) == 0 ? imageLoad(specular_ping, q) : imageLoad(specular_pong, q);
+                const float lum_q = rgb_to_luminance(color_q.rgb);
+                const float z_q = imageLoad(in_linear_depth, q).r;
+                const vec3 n_q = unpack_gbuffer_normal(imageLoad(in_normal, q).rg);
+                const float roughness_q = imageLoad(in_material, q).y;
+
+                float w_n = pow(max(0.0, dot(n_q, n_center)), control.sigma_n);
+                float w_z = abs(z_q - z_center.x) / (control.sigma_z * abs(dot(depth_grad, p.xy - q.xy)) + 1e-6);
+                float w_l = abs(lum_q - center_luminance) / phi_color;
+                float w = w_n * exp(-max(0.0, w_z) - max(0.0, w_l));
+
+                float r_max = 0.1;
+                float r_near = 0.01;
+                float r_far = 0.1;
+                float d_near = 10.0 * roughness_center;
+                float d_far = 70.0 * roughness_center;
+                float w_r = 1.0 - smoothstep(r_near, r_far, abs(roughness_q - roughness_center));
+                float w_d = 1.0 - smoothstep(d_near, d_far, length(vec3(p - q)));
+
+                float w_spec = exp(-max(0.0, w_z)) * w_n * w_r * w_d;
```

Comment:
It looks like variance won't affect the specular filtering, right? What would happen if you used a coefficient like `w_l` here too, but calculated its `phi_color` from the specular images?